common/src/Microsoft.Azure.IIoT.Serializers.NewtonSoft/src/Default/NewtonSoftJsonSerializer.cs

```diff
@@ -72,20 +74,36 @@ public class NewtonSoftJsonSerializer : IJsonSerializerSettingsProvider,
             try {
                 // TODO move to .net 3 to use readonly span as stream source
                 var jsonSerializer = JsonSerializer.CreateDefault(Settings);
-                using (var stream = new MemoryStream(buffer.ToArray()))
+                var bufferArray = buffer.ToArray();
+                using (var stream = new MemoryStream(bufferArray))
                 using (var reader = new StreamReader(stream, ContentEncoding)) {
 
                     // Validate json if schema is provided.
                     if (schemaReader != null) {
-                        var messages = new List<string>();
-                        using (var jsonSchemaReader = new JsonTextReader(schemaReader)) {
-                            var schema = JSchema.Load(jsonSchemaReader);
-                            using (var validatingReader = new JSchemaValidatingReader(new JsonTextReader(reader))) {
-                                validatingReader.Schema = schema;
-                                validatingReader.ValidationEventHandler += (o, a) => messages.Add(a.Message);
-                                var serializedType = jsonSerializer.Deserialize(validatingReader, type);
-
-                                return messages.Count > 0 ? throw new JsonSerializationException(string.Join(", ", messages.ToArray())) : serializedType;
+                        using (var validationDataStream = new MemoryStream(bufferArray))
+                        using (var validationReader = new StreamReader(validationDataStream)) {
+                            var schema = JsonSchema.FromText(schemaReader.ReadToEnd());
+                            var validationOptions = new ValidationOptions() {
+                                OutputFormat = OutputFormat.Detailed,
+                            };
+                            var validationResults = 
+                                schema.Validate(
+                                    JsonDocument.Parse(
+                                        validationDataStream.ReadAsString(Encoding.UTF8), 
+                                        new JsonDocumentOptions() { AllowTrailingCommas = true, }
+                                        ).RootElement,
+                                    validationOptions);
+                            var serializedType = jsonSerializer.Deserialize(reader, type);
```

Comment:
I think you probably don't want to deserialize json into object at this stage, only if validationResults.IsValid == true?

Reply:
Yeah good point. I'll make the change.